components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -3584,334 +3576,608 @@ end subroutine mam_soaexch_1subarea
-         dryvol_smallest(mfrm) = 1.0e-25
-!        v2nlorlx(mfrm) = voltonumblo_amode(mfrm)*frelax
-!        v2nhirlx(mfrm) = voltonumbhi_amode(mfrm)/frelax
-         v2nlorlx(mfrm) = ( 1._r8 / ( (pi/6._r8)* &
-            (dgnumlo_aer(mfrm)**3._r8)*exp(4.5_r8*alnsg_aer(mfrm)**2._r8) ) ) * frelax
-         v2nhirlx(mfrm) = ( 1._r8 / ( (pi/6._r8)* &
-            (dgnumhi_aer(mfrm)**3._r8)*exp(4.5_r8*alnsg_aer(mfrm)**2._r8) ) ) / frelax
-
-         tmp_alnsg2(mfrm) = 3.0 * (alnsg_aer(mfrm)**2)
-         dp_cut(mfrm) = sqrt(   &
-            dgnum_aer(mfrm)*exp(1.5*(alnsg_aer(mfrm)**2)) *   &
-            dgnum_aer(mtoo)*exp(1.5*(alnsg_aer(mtoo)**2)) )
-         lndp_cut(mfrm) = log( dp_cut(mfrm) )
-         dp_belowcut(mfrm) = 0.99*dp_cut(mfrm)
-      end do
-      if (npair <= 0) return
+        !intent-outs
+        integer,  intent(out) :: num_pairs         ! total number of pairs to be found
+        real(r8), intent(out) :: sz_factor(:), fmode_dist_tail_fac(:) !precomputed factors to be used later
+        real(r8), intent(out) :: v2n_lo_rlx(:), v2n_hi_rlx(:)         !relaxed volume to num high and low ratio limits
+        real(r8), intent(out) :: ln_diameter_tail_fac(:)              !log of diameter factor for distribution tail
+        real(r8), intent(out) :: diameter_cutoff(:), ln_dia_cutoff(:) !cutoff (threshold) for deciding the  do inter-mode transfer
+        real(r8), intent(out) :: diameter_threshold(:)                !threshold diameter(99% of cuttoff)
 
-! compute aerosol dry-volume for the "from mode" of each renaming pair
-! also compute dry-volume change during the continuous growth process
-      do n = 1, ntot_amode
-         mtoo = mtoo_renamexf(n)
-         if (mtoo <= 0) cycle
+        !local variables
+        integer :: dest_mode, src_mode, imode
 
-         tmpa = 0.0_r8 ; tmpb = 0.0_r8
-         do iaer = 1, naer
-!   fac_m2v_aer converts (kmol-AP/kmol-air) to (m3-AP/kmol-air)
-            tmpa = tmpa + qaer_cur(iaer,n)*fac_m2v_aer(iaer)
-            tmpb = tmpb + qaer_del_grow4rnam(iaer,n)*fac_m2v_aer(iaer)
-         end do
-         dryvol_a(n) = tmpa-tmpb ! dry volume before growth
-         deldryvol_a(n) = tmpb   ! change to dry volume due to growth
+        !some parameters
+        real(r8), parameter :: sqrt_half = sqrt(0.5)
+        real(r8), parameter :: frelax = 27.0_r8 !(3^3)
 
-         if ( iscldy_subarea ) then
-         tmpa = 0.0_r8 ; tmpb = 0.0_r8
-         do iaer = 1, naer
-!   fac_m2v_aer converts (kmol-AP/kmol-air) to (m3-AP/kmol-air)
-            tmpa = tmpa + qaercw_cur(iaer,n)*fac_m2v_aer(iaer)
-            tmpb = tmpb + qaercw_del_grow4rnam(iaer,n)*fac_m2v_aer(iaer)
-         end do
-         dryvol_c(n) = tmpa-tmpb 
-         deldryvol_c(n) = tmpb
-         end if ! ( iscldy_subarea ) then
+        !number of pairs allowed to do inter-mode particle transfer
+        ! (e.g. if we have a pair "mode_1<-->mode_2", mode_1 and mode_2 can participate in
```

Comment:
Great explanation here. If you can copy something from this to the above location, a developer will be able to scan the subroutine more easily.